Data/Base/Shaders/Materials/MaterialHelper.h

```diff
@@ -88,36 +88,48 @@ ezMaterialData FillMaterialData()
     matData.worldPosition = float3(0.0, 0.0, 0.0);
   #endif
 
-  float3 worldNormal = normalize(GetNormal());
-  #if TWO_SIDED == TRUE && defined(USE_TWO_SIDED_LIGHTING)
-    #if FLIP_WINDING == TRUE
-      matData.worldNormal = G.Input.FrontFace ? -worldNormal : worldNormal;
-    #else
-      matData.worldNormal = G.Input.FrontFace ? worldNormal : -worldNormal;
-    #endif
-  #else
+#if defined(USE_NORMAL)
```

Comment:
Technically this is not correct: USE_NORMAL indicates whether you want to use the **mesh** normal stream and interpolator. You can still have a GetNormal function that procedurally generates normals but doesn't use the mesh normal stream.